my-app/src/app/core/service/loading.service.ts

```diff
@@ -4,8 +4,25 @@ import { BehaviorSubject } from 'rxjs';
 @Injectable({
   providedIn: 'root'
 })
+
 export class LoadingService {
+  loadingSub: BehaviorSubject<boolean> = new BehaviorSubject<boolean>(false);
+  public loadingMap: Map<string, boolean> = new Map<string, boolean>();
```

Comment:
1) you don't need this map. 
2) don't create such public properties and edit it outside the service. It violates the encapsulation principle